### ISSUE TITLE
it's not working with axios v.19.0

### ISSUE THREAD

**Issue by xnog** [JustinBeckwith/retry-axios#62]
it's not working with axios v.19.0

**Comment by dennisreimann** [comment]
Yes, the reason for that is – as @adntin [points out here](https://github.com/axios/axios/issues/164#issuecomment-508367282) – that Axios 0.19 filters out custom properties.

**Comment by JustinBeckwith** [comment]
Yeah - sadly I doubt I'm going to be updating this library to support the latest version 😢 We just don't use this at my company anymore, so I don't have a ton of time to keep it up to date.  If others are interested ...  I'd happily accept PRs!

**Comment by dennisreimann** [comment]
fyi: We just fixed this by using the same meachanism, but using a custom header field instead of the custom config property.

**Comment by avindra** [comment]
This is kind of janky but you can patch axios 0.19 directly in node_modules to permit `raxConfig` as a valid key.

Patch:

```diff
diff --git a/node_modules/axios/lib/core/mergeConfig.js b/node_modules/axios/lib/core/mergeConfig.js
index 6097a3e..1ec151e 100644
--- a/node_modules/axios/lib/core/mergeConfig.js
+++ b/node_modules/axios/lib/core/mergeConfig.js
@@ -38,7 +38,7 @@ module.exports = function mergeConfig(config1, config2) {
     'timeout', 'withCredentials', 'adapter', 'responseType', 'xsrfCookieName',
     'xsrfHeaderName', 'onUploadProgress', 'onDownloadProgress', 'maxContentLength',
     'validateStatus', 'maxRedirects', 'httpAgent', 'httpsAgent', 'cancelToken',
-    'socketPath'
+    'socketPath', 'raxConfig'
   ], function defaultToConfig2(prop) {
     if (typeof config2[prop] !== 'undefined') {
       config[prop] = config2[prop];
```

then in your package.json:

```json
    "postintall": "patch -p1 -N < /axios-config-fix.patch",
```

**Comment by KayHS** [comment]
@dennisreimann Hi please can you demonstrate how you managed to fix this?

**Comment by dennisreimann** [comment]
@KayHS see [this comment](https://github.com/axios/axios/issues/164#issuecomment-524756674).

Whereas before you have assigned the property directly to the config (i.e. `config.retryCount`) you now need to use a custom header for that (i.e. `headers.common['retryCount']`).

**Comment by KayHS** [comment]
@dennisreimann 

hnn, my setup is like the following

```
import { attach, getConfig } from "retry-axios";
import axios from "axios";

 const interceptorId = attach();

  const options: any = {
            method: "get",
            url,
            headers: {
                "X-ID": classificationRequest.id,
            },
            data: {},
            raxConfig: retryConfig,
        };
```
so headers.common['raxConfig'] = retryConfig?

**Comment by dennisreimann** [comment]
Or something like this:

```js
headers: {
  "X-ID": classificationRequest.id,
  "x-retry-count": 0
},
```

It does not really matter, as long as you write to the same header property you also read from.

**Comment by findmory** [comment]
or you can use the master version on git `npm install axios/axios#master` . I've confirmed that they fixed the issue with custom params there just haven't pushed it to npm

**Comment by fabrusso** [comment]
> or you can use the master version on git `npm install axios/axios#master` . I've confirmed that they fixed the issue with custom params there just haven't pushed it to npm

Tried it and It's working, also with other NPM libraries for retries.

Thanks a lot.

**Comment by DanielRuf** [comment]
You can apply the patch using `patch-package`.

**Comment by DanielRuf** [comment]
It is still broken in 0.19.1 with the default config.

**Comment by JustinBeckwith** [comment]
@DanielRuf it looks like `0.19.1` went out, and _should_ include the fix.  I am a little confused because my tests currently use `0.19.1`, and everything appears to be passing.  Could I trouble someone for a simple repro that shows this?

**Comment by DanielRuf** [comment]
In our project it didn't work with the custom options. Headers might work but I chose the normal solution from the docs.

**Comment by DanielRuf** [comment]
https://github.com/axios/axios/compare/v0.19.0...v0.19.1

**Comment by DanielRuf** [comment]
https://github.com/axios/axios/commit/a11cdf468303a365a6dc6e84f6dd0e4b3b8fd336

**Comment by DanielRuf** [comment]
Seems to be fixed in 0.19.1 but we'll still apply the patch to ensure that it will work.

**Comment by JustinBeckwith** [comment]
Awesome.  Seems like it's working, so I'm closing this out :)
